zkvm/src/vm.rs

```diff
@@ -360,16 +360,18 @@ where
         Ok(())
     }
 
+    /// _qty flv data pred_ **issue** → _contract_
     fn issue(&mut self) -> Result<(), VMError> {
         let predicate = self.pop_item()?.to_data()?.to_predicate()?;
+        let metadata = self.pop_item()?.to_data()?;
         let flv = self.pop_item()?.to_variable()?;
         let qty = self.pop_item()?.to_variable()?;
 
         let (flv_point, _) = self.attach_variable(flv)?;
         let (qty_point, _) = self.attach_variable(qty)?;
 
         self.delegate.verify_point_op(|| {
-            let flv_scalar = Value::issue_flavor(&predicate);
+            let flv_scalar = Value::issue_flavor(&predicate, &metadata);
```

Comment:
per comment above, `metadata` can be moved into `issue_flavor` here to avoid unnecessary allocations